subsys/bluetooth/controller/hci_driver.c

```diff
@@ -409,19 +409,20 @@ static const struct bt_hci_driver drv = {
 	.send = hci_driver_send,
 };
 
-uint8_t bt_read_static_addr(struct bt_hci_vs_static_addr *addr)
+uint8_t bt_read_static_addr(struct bt_hci_vs_static_addr addrs[], uint8_t size)
 {
-	if (((NRF_FICR->DEVICEADDR[0] != UINT32_MAX) ||
-	    ((NRF_FICR->DEVICEADDR[1] & UINT16_MAX) != UINT16_MAX)) &&
-	     (NRF_FICR->DEVICEADDRTYPE & 0x01)) {
-		sys_put_le32(NRF_FICR->DEVICEADDR[0], &addr->bdaddr.val[0]);
-		sys_put_le16(NRF_FICR->DEVICEADDR[1], &addr->bdaddr.val[4]);
+	if ((size > 0) &&
```

Comment:
only one address is read?

Reply:
implementation extension is up to the controller developers

Reply:
So let's ping them. @rugeGerritsen

Reply:
We only need to support one address, which is also how it is done for the Zephyr LL. The zephyr one is slightly updated, so you may consider replacing it with: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/hci/nordic/hci_vendor.c#L14

Reply:
+1 on re-using `hci_vendor_read_static_addr`

Reply:
implementation replaced